### PR TITLE
fix: don't try to create a bucket if it does not exist

### LINKED ISSUE #53: The sink should not call create bucket
After the change of bubbling up the error on creating a bucket it exposed the real error for us - the user does not have permission to create a bucket.

We should just try to use `bucket` and bubble up if it errors

### PR DESCRIPTION
**Pull request by SimenB**
## Status
**READY**

## Description
Fixes #53

We never got an error because the bucket did not exist, but because the user did not have the correct permissions. This drops the attempted cleverness in handling it, and just bubble up the error to the user. It'll look like

```
{ ApiError: Not Found
    at Object.parseHttpRespBody (/Users/simbekkh/repos/podium-asset-server/node_modules/@google-cloud/common/src/util.js:192:30)
    at Object.handleResp (/Users/simbekkh/repos/podium-asset-server/node_modules/@google-cloud/common/src/util.js:132:18)
    at /Users/simbekkh/repos/podium-asset-server/node_modules/@google-cloud/common/src/util.js:465:12
    at Request.onResponse [as _callback] (/Users/simbekkh/repos/podium-asset-server/node_modules/retry-request/index.js:182:7)
    at Request.self.callback (/Users/simbekkh/repos/podium-asset-server/node_modules/request/request.js:186:22)
    at emitTwo (events.js:126:13)
    at Request.emit (events.js:214:7)
    at Request.<anonymous> (/Users/simbekkh/repos/podium-asset-server/node_modules/request/request.js:1163:10)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
  code: 404,
  errors:
   [ { domain: 'global', reason: 'notFound', message: 'Not Found' } ],
  response: undefined,
  message: 'Not Found' }
```

## Related PRs
* None

### CODE REVIEW

File: lib/sink.js [hunk 1]
```diff
@@ -141,29 +141,17 @@ module.exports = class SinkGCS extends EventEmitter {
...
+        const bucket = this.gcs.bucket(bucketName);
 
-                return Promise.reject(error);
-            });
+        // `getMetadata` fails if the bucket does not exist
+        this._ready = bucket.getMetadata().then(() => {
+            this.bucket = bucket;
+            this.emit(
+                'storage info',
+                `Bucket "${bucketName}" did not exist. Created bucket`
```
Comment (sveisvei) [inline lib/sink.js]: This message is wrong, ey?

Comment (SimenB) [reply lib/sink.js]: yup, just pushed a fix for it

File: lib/sink.js [hunk 2]
```diff
@@ -141,29 +141,17 @@ module.exports = class SinkGCS extends EventEmitter {
...
-                        error
-                    );
-                    return this.bucket;
-                }
+        const bucket = this.gcs.bucket(bucketName);
 
-                return Promise.reject(error);
-            });
+        // `getMetadata` fails if the bucket does not exist
+        this._ready = bucket.getMetadata().then(() => {
```
Comment (trygve-lie) [inline lib/sink.js]: Symbols to ensure proper privacy.

**Comment by SimenB** [comment]
>So then at the app level we handle this case and log appropriately?

We crash. Maybe not ideal, but I don't think anything else really makes sense - the sink is unusable

**Comment by digitalsadhu** [comment]
so its then the responsibility of the consumer to ensure that the correct bucket is in place before use I guess?

**Comment by SimenB** [comment]
>so its then the responsibility of the consumer to ensure that the correct bucket is in place before use I guess?

Yup, that was my thinking

**Pull request merged**